grant/projects/offchain-subnames.md

```diff
@@ -0,0 +1,137 @@
+
+It leverages the EIP-3668 specification to enable efficient and cost-effective management of subname-address mappings off the Ethereum mainnet. By storing these mappings on Darwinia Chain, the project offers a scalable approach to handle a large number of subdomains without incurring high gas fees.
+
+The solution utilizes a customized OffchainResolver contract on Ethereum, which works with a gateway service connected to the Darwinia Chain. This setup allows for seamless resolution of subnames (e.g., 'foo.ringdao.eth') while maintaining the security and decentralization benefits of the ENS.
+
+<!-- Key features of the project include:
+1. Offchain storage of subname-address mappings on Darwinia Chain
+2. Integration with ENS on Ethereum using EIP-3668 for secure offchain data retrieval
+3. A gateway service to facilitate communication between Ethereum and Darwinia Chain
+4. User-friendly UIs for subname registration, update, and resolution.
+5. Admin UI for subname management. -->
+
+This project addresses the need for efficient subdomain management within the Darwinia ecosystem, as outlined in the RFP (https://github.com/ringecosystem/collaboration/issues/61), and demonstrates the potential for cross-chain interoperability in domain name services.
+
+### This project will benefit the Darwinia ecosystem in the following ways:
+
+1. It will provide a more efficient and cost-effective way to manage subdomains on the Darwinia Chain, reducing the need for users to pay high gas fees on Ethereum.
+2. It will enhance the usability of the 'ringdao.eth' domain by allowing users to register, update, and resolve subnames with reduced costs and improved performance.
+3. It will demonstrate the potential for cross-chain interoperability in domain name services, opening up new possibilities for collaboration and innovation within the Darwinia ecosystem.
+
+## Team
+
+- Aki Wu([wuminzhe](https://github.com/wuminzhe))  
+  A full stack developer focused on the web3 space. He has been working on darwinia ecosystem for a long time.
+
+- Echo([Echo](https://github.com/hujw77))  
+  Contracts developer from Itering.
+
+
+## Team Code Repos
+
+- https://github.com/wuminzhe/basenames
+  The repo for the basenames contracts.
+
+- https://github.com/wuminzhe/offchain-resolver  
+  1. the offchain gateway.
+  2. the offchain resolve client.
+
+- https://github.com/wuminzhe/subnames-ui
+  The repo for the UI.
+
+## Development Roadmap
+
+### Milestone 1
+
+Estimated Duration: 2 weeks
+
+Cost: 0.25 ETH
+
+Address: 0x2Da8ccfe0dD2165B8d939eaBf4E3697C4Adb6FDd
+
+#### Tasks:
+
+1. Contracts pre research & implementation.  
+2. Adjust the contracts to align with Darwinia Chain.
+3. Deploy to testnet(Darwinia Koi).
+
+Deliverables:
+
+- The contracts source code in MIT license.
+- The deployed contracts address.
+
+### Milestone 2
+
+Estimated Duration: 2 weeks
+
+Cost: 0.25 ETH
```

Comment:
Due to significant market fluctuations, please use USDC as the unit. As previously mentioned, the total sponsorship amount will be approximately 2K USD, with a potential increase to 2.5K USD if the committee deems the deliverables to be outstanding.

Reply:
My price was calculated in eth from the beginning. For full workload, I originally hope to get 3eth, but you said you hope this is a business project of my own, so 1 ETH is the minimum I hope to get to start this project.

Reply:
Thank you for your prompt response. Due to Grant policy restrictions and limited budget, we regret that we are unable to fulfill your sponsorship request to support this Grant. We look forward to future collaboration opportunities.

Reply:
No problem. It took me a long time to do research and development. And If I can not get enough confidence from the first deal, that means this is not a good startup project with strong demand. That's why I insist on this price from beginning.  

Contact me if needs or ringdao's police changed. I'll look to other opportunities to collaborate with ringdao.

Reply:
Sorry for having ignored some history messages, after I checked the history, I saw you have confirmed the price in the this thread before here. If you approved, I will update the price to 2.5k and continue this PR. 

Why 2.5k? Basenames itself has not provide the frontend functionality of extend/renew. But I have done it(extend, but renew is very similar). That shows my effortness to hope do it well. You can try it here: https://subnames-ui.vercel.app/, after you input your registered subname, it will tell you its expiry and the extend flow.